python/vineyard/core/client.py

```diff
@@ -60,18 +62,25 @@ def __init__(
         port: Union[int, str] = None,
         # move host after port to make sure unnamed (host, port) works
         host: str = None,
-        endpoint: Tuple[str, Union[str, int]] = None,
+        endpoint: Union(str, Tuple[str, Union[str, int]]) = None,
         session: int = None,
         username: str = None,
         password: str = None,
+        config: str = None,
     ):
         """Connects to the vineyard IPC socket and RPC socket.
 
-        - The argument `socket` takes precedence over environment variable
-          `VINEYARD_IPC_SOCKET` for IPC client.
-        - The argument `endpoint` takes precedence over arguments (`host`, `port`),
-          which further takes precedence over environment variable
-          `VINEYARD_RPC_ENDPOINT` for RPC client.
+        The precedence of the arguments are as follows:
+        - Argument `socket`  > Env `VINEYARD_IPC_SOCKET` > Config `IPCSocket`
+        - Argument `endpoint` > Env `VINEYARD_RPC_ENDPOINT` > Config `RPCEndpoint`
+
+        This parameter `config` can either be a path to a YAML configuration file or
+        a path to a directory containing the default config file `vineyard.yaml`.
+        The content of the configuration file should look like this:
+
+        Vineyard:
+          IPCSocket: '/path/to/vineyard.sock'
+          RPCEndpoint: 'hostname1:port1,hostname2:port2,...'
```

Comment:
For source code block in comments, use 

```
.. code:: yaml

   ...
```

to make sure it looks nice in documentation.